operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResource.java

```diff
@@ -12,4 +13,46 @@ default EventSource initEventSource(EventSourceContext<P> context) {
   default Class<R> resourceType() {
     return (Class<R>) Utils.getFirstTypeArgumentFromInterface(getClass());
   }
+
+  /**
+   * todo: return Optional<R></R> instead?
+   * 
+   * Computes the desired state of the dependent based on the state provided by the specified
+   * primary resource.
+   * 
+   * The default implementation returns {@code null} which corresponds to the case where the
+   * associated dependent should never be created by the associated reconciler or that the global
+   * state of the cluster doesn't allow for the resource to be created at this point.
+   * 
+   * @param primary the primary resource associated with the reconciliation process
+   * @param context the {@link Context} associated with the reconciliation process
+   * @return an instance of the dependent resource matching the desired state specified by the
+   *         primary resource or {@code null} if the dependent shouldn't be created at this point
+   *         (or ever)
+   */
+  default R desired(P primary, Context context) {
+    return null;
+  }
+
+  /**
+   * Checks whether the actual resource as fetched from the cluster matches the desired state
+   * expressed by the specified primary resource.
+   * 
+   * The default implementation always return {@code true}, which corresponds to the behavior where
+   * the dependent never needs to be updated after it's been created.
+   * 
+   * Note that failure to properly implement this method will lead to infinite loops. In particular,
+   * for typical Kubernetes resource implementations, simply calling
+   * {@code desired(primary, context).equals(actual)} is not enough because metadata will usually be
+   * different.
+   * 
+   * @param actual the current state of the resource as fetched from the cluster
+   * @param primary the primary resource associated with the reconciliation request
+   * @param context the {@link Context} associated with the reconciliation request
+   * @return {@code true} if the actual state of the resource matches the desired state expressed by
+   *         the specified primary resource, {@code false} otherwise
+   */
+  default boolean match(R actual, P primary, Context context) {
+    return true;
```

Comment:
shouldn't we use `equals` here?

Reply:
No, see the javadoc 😉

Reply:
```
for typical Kubernetes resource implementations, simply calling
   * {@code desired(primary, context).equals(actual)} is not enough because metadata will usually be
   * different.
```
This is true, but in a default implementation I would just compare specs. And in reconcile, would merge the target resource with the actual from the server. Thus just override the spec, add other fields if new to metadata (and override if some is generated). Would also make this behavior customizable (already is, some can override), to make a custom comaparator.

```
  * The default implementation always return {@code true}, which corresponds to the behavior where
   * the dependent never needs to be updated after it's been created.   
```
Why should be this the default?

Reply:
> This is true, but in a default implementation I would just compare specs. And in reconcile, would merge the target resource with the actual from the server. Thus just override the spec, add other fields if new to metadata (and override if some is generated). Would also make this behavior customizable (already is, some can override), to make a custom comaparator.
 
That's a good point

> Why should be this the default?

Why shouldn't it? 😄 
More seriously what do you think the default behavior should be? Not updating the resource after it's been created sounds like a reasonable default which covers a lot of use cases (for example, pretty much all our samples), imo.

Reply:
I think it should not have a default implementation. 

The most common case is like in the WebPage, when all the resources are managed by the operator, IMHO that should be the default approach. If the resource is not managed by us (so not created and not updated) should be probably a feature flag/mode on the resources. Maybe eventually a different type of resource.

Reply:
The default should be to generate the target default with desired, and compare the specs. 

The reason is that I think the default use case should be what we have in WebPage (what actally needs some cleanup), but basically that the reconciler manages all the resources. That is not that strong assumption IMHO, sure there are cases where not, but this would be quite sensible default.